modules/imgproc/src/imgwarp.cpp

```diff
@@ -1816,6 +1815,10 @@ void cv::remap( InputArray _src, OutputArray _dst,
     if( dst.data == src.data )
         src = src.clone();
 
+    if ((map1.type() == CV_32FC1) && (map2.type() == CV_32FC1))
+        CALL_HAL(remap, cv_hal_remap, src.type(), src.data, src.step, src.cols, src.rows, dst.data, dst.step, dst.cols, dst.rows,
```

Comment:
In general HAL API should be called unconditionally.

Or we need to rename API and apply `32f` suffix.